pkg/dns/aws/dns_test.go

```diff
@@ -0,0 +1,218 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/client"
+	"github.com/aws/aws-sdk-go/aws/client/metadata"
+	"github.com/aws/aws-sdk-go/aws/request"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/elb"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/route53"
+
+	configv1 "github.com/openshift/api/config/v1"
+
+	iov1 "github.com/openshift/api/operatoringress/v1"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"testing"
+)
+
+func getResponse(record iov1.DNSRecord, dnsZone configv1.DNSZone) []interface{} {
+	var elbDescriptions []*elb.LoadBalancerDescription
+	var responses []interface{}
+
+	for _, target := range record.Spec.Targets {
+		elbDescription := &elb.LoadBalancerDescription{
+			DNSName:                   aws.String(target),
+			CanonicalHostedZoneNameID: aws.String(dnsZone.ID),
+		}
+		elbDescriptions = append(elbDescriptions, elbDescription)
+
+		elbResponse := &elb.DescribeLoadBalancersOutput{
+			LoadBalancerDescriptions: elbDescriptions,
+		}
+
+		responses = append(responses, elbResponse)
+	}
+
+	route53Response := &route53.ChangeResourceRecordSetsInput{}
+	responses = append(responses, route53Response)
+
+	return responses
+}
+
+func FakeAWSProvider(resps []interface{}) *Provider {
+	reqNum := 0
+	sess := session.Must(session.NewSession(&aws.Config{
+		DisableSSL: aws.Bool(true),
+	}))
+	sess.Handlers.Send.Clear()
+	sess.Handlers.Unmarshal.Clear()
+	sess.Handlers.UnmarshalMeta.Clear()
+	sess.Handlers.ValidateResponse.Clear()
+	sess.Handlers.Unmarshal.PushBack(func(r *request.Request) {
+		r.Data = resps[reqNum]
+		reqNum++
```

Comment:
Every caller to `FakeAWSProvider` follows this pattern:
```go
	responses := getResponse(record, dnsZone)
￼	mgr := FakeAWSProvider(responses)
```
I think you could instead pass `record` and `dnsZone` to `FakeAWSProvider`, and then you could call `getResponse` here:
```go
	resps := getResponse(record, dnsZone)
	sess.Handlers.Unmarshal.PushBack(func(r *request.Request) {
		r.Data = resps[reqNum]
￼		// ...
```

Also, the name `getResponse` could be clearer; for example, `buildFakeResponses` makes it a little more obvious that (a) the function is constructing the responses, not getting them from somewhere else; (b) the responses are fake; and (c) it is a slice of responses, not a single response.